test/e2e/e2e_suite_test.go

```diff
@@ -17,104 +16,216 @@ limitations under the License.
 package e2e
 
 import (
+	"context"
 	"fmt"
+	"log"
 	"os"
-	"os/exec"
 	"testing"
 
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/envfuncs"
+	"sigs.k8s.io/e2e-framework/support/kind"
+	"sigs.k8s.io/e2e-framework/support/utils"
 
-	"go.etcd.io/etcd-operator/test/utils"
+	kubebuilder_utils "go.etcd.io/etcd-operator/test/utils"
 )
 
 var (
-	// Optional Environment Variables:
-	// - PROMETHEUS_INSTALL_SKIP=true: Skips Prometheus Operator installation during test setup.
-	// - CERT_MANAGER_INSTALL_SKIP=true: Skips CertManager installation during test setup.
-	// These variables are useful if Prometheus or CertManager is already installed, avoiding
-	// re-installation and conflicts.
-	skipPrometheusInstall  = os.Getenv("PROMETHEUS_INSTALL_SKIP") == "true"
-	skipCertManagerInstall = os.Getenv("CERT_MANAGER_INSTALL_SKIP") == "true"
-	// isPrometheusOperatorAlreadyInstalled will be set true when prometheus CRDs be found on the cluster
-	isPrometheusOperatorAlreadyInstalled = false
-	// isCertManagerAlreadyInstalled will be set true when CertManager CRDs be found on the cluster
-	isCertManagerAlreadyInstalled = false
-
-	// projectImage is the name of the image which will be build and loaded
-	// with the code source changes to be tested.
-	projectImage = "example.com/etcd-operator:v0.0.1"
+	testEnv     env.Environment
+	dockerImage = "etcd-operator-controller:v0.1"
+	namespace   = "etcd-operator-system"
 )
 
-// TestE2E runs the end-to-end (e2e) test suite for the project. These tests execute in an isolated,
-// temporary environment to validate project changes with the the purposed to be used in CI jobs.
-// The default setup requires Kind, builds/loads the Manager Docker image locally, and installs
-// CertManager and Prometheus.
-func TestE2E(t *testing.T) {
-	RegisterFailHandler(Fail)
-	_, _ = fmt.Fprintf(GinkgoWriter, "Starting etcd-operator integration test suite\n")
-	RunSpecs(t, "e2e suite")
+func TestMain(m *testing.M) {
+	testEnv = env.New()
+	kindClusterName := "etcd-cluster"
+	kindCluster := kind.NewCluster(kindClusterName)
+
+	log.Println("Creating KinD cluster...")
+	origWd, _ := os.Getwd()
+	testEnv.Setup(
+		// create KinD cluster and namespace
+		envfuncs.CreateCluster(kindCluster, kindClusterName),
+		envfuncs.CreateNamespace(namespace),
+
+		// prepare the resources
+		func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+			// change dir for Make file
+			if err := os.Chdir("../../"); err != nil {
+				log.Printf("Unable to set working directory: %s", err)
+				return ctx, err
+			}
```

Comment:
We expect users to run the e2e test from the root directory. `go test ./test/e2e/`
```suggestion
```